ArduCopter/mode_auto.cpp

```diff
@@ -1617,6 +1623,14 @@ bool ModeAuto::set_next_wp(const AP_Mission::Mission_Command& current_cmd, const
         get_spline_from_cmd(next_cmd, default_loc, next_dest_loc, next_next_dest_loc, next_next_dest_loc_is_spline);
         return wp_nav->set_spline_destination_next_loc(next_dest_loc, next_next_dest_loc, next_next_dest_loc_is_spline);
     }
+    case MAV_CMD_NAV_ARC_WAYPOINT: {
+        const Location dest_loc = loc_from_cmd(current_cmd, default_loc);
+        const Location next_dest_loc = loc_from_cmd(next_cmd, dest_loc);
+        // We had to do some special handling to store turn direction/sign of arc_angle as next_cmd.p1 is a uint16_t
+        const float sign = next_cmd.content.location.loiter_ccw == 0 ? 1.0 : -1.0;
+        const float arc_angle = radians(float(next_cmd.p1) * sign);
```

Comment:
Better to add a helper function to keep the mission weirdness in mission.  Something like `get_arc_angle_rad`. Same idea as a existing one for loiter turns.

https://github.com/ArduPilot/ardupilot/blob/b876809242aebe3d9eef87b50ea8076fadcb6e06/libraries/AP_Mission/AP_Mission.h#L426-L437